figures/mapping-histology-labels.Rmd

```diff
@@ -0,0 +1,237 @@
+
+```{r}
+small_groups <- broad_summary %>% 
+  dplyr::filter(n < small_groups_cutoff) %>% 
+  dplyr::pull(broad_histology)
+```
+
+Let's take a look at how big all these groups are with a barplot.
+We'll mark which small histology groups are going to be labeled as "Other". 
+
+```{r}
+broad_summary %>% 
+  dplyr::mutate(now_labeled_other = broad_histology %in% small_groups) %>% 
+ggplot2::ggplot(ggplot2::aes(x = reorder(broad_histology, -n), y = n, fill = now_labeled_other)) +
+  ggplot2::geom_bar(position = "dodge", stat = "identity") +
+  ggplot2::theme_classic() +
+  ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 1)) + 
+  ggplot2::xlab("broad_histology label") +
+  ggplot2::ggtitle(paste("Broad histologies: groups <", small_groups_cutoff, "will be 'Other' in `summary_group`"))
+```
+
+There's handful of very small groups (many are n = 2). 
+
+# LGAT adjacent groups 
+
+Previously samples with these labels were generally considered `LGAT` groups, so we will group them back in in `summary_group`. 
+
+```{r}
+lgat_adjacent <- c("Neuronal and mixed neuronal-glial tumor", 
+                   "Diffuse astrocytic and oligodendroglial tumor")
+```
+
+Stop if these strings aren't detected. 
+
+```{r}
+if (!all(lgat_adjacent %in% metadata$broad_histology)) {
+  stop("LGAT adjacent groups' text strings are not detected in this version of pbta-histologies' broad_histology variable")
+}
+```
+
+# Make new `summary_group`
+
+This [logic is described at the top of the notebook](#how_summary_group_is_made:).
+
+```{r}
+histology_table <- metadata %>% 
+  dplyr::mutate(
+    summary_group = dplyr::case_when(
+      broad_histology %in% small_groups ~ "Other",
+      broad_histology %in% lgat_adjacent ~ "Low-grade astrocytic tumor", 
+      sample_type == "Normal" ~ "Normal",
+      TRUE ~ broad_histology)
+    )
+```
+
+Print out the number of `summary_groups` (including `Normal`)!
+
+```{r}
+summary_groups <- histology_table %>% 
+  dplyr::count(summary_group)
+
+nrow(summary_groups)
+```
+
+Make this notebook stop if there are more than 15 groups. 
+
+```{r}
+if (nrow(summary_groups) > 15) {
+  stop("There's more than 15 `summary_groups` may want to re-evaluate the high-level histology groupings")
+}
+```
+
+# Add hex codes 
+
+These hex codes were retrieved from http://phrogz.net/css/distinct-colors.html with the settings on default for 18 colors.
+
+```{r}
+color_palette <- 
+  c("#ff0000", "#cc0000", "#995200", "#bfb300", "#fffbbf", 
+    "#2e7300", "#00e65c", "#00ffee", "#103d40", "#0085a6", 
+    "#003380", "#4073ff", "#737899", "#70008c", "#f2b6ee", 
+    "#ff40bf", "#8c0038", "#330d12"
+)
+```
+
+Declare how many colors we need. 
+
+```{r}
+n_colors <- nrow(summary_groups)
+```
+
+Make a named list color key where histologies are the names. 
+
+```{r}
+# Set seed so the colors are consistent upon re-run
+set.seed(2021)
+
+# Sample from the 18 colors
+subset_colors <- sample(color_palette, n_colors)
```

Comment:
Alternatively, we could drop this "Other" group from the large summary figures.

Reply:
We consider treating "Benign" as the grey group @jashapiro and @cansavvy - because it's more like a control and we would be less interested (maybe) in those genomics.